src/cli/api/apihelper.go

```diff
@@ -379,6 +383,77 @@ func (helper *APIHelper) GetMetrics(metricName string, startTime, endTime int64,
 
 }
 
+func (helper *APIHelper) GetAggregatedMetrics(metricName string, startTime, endTime int64, desc bool, page uint64) (bool, [][]string, error) {
+
+	hasNext := false
+	if page <= 1 {
+		err := helper.CheckHealth()
+		if err != nil {
+			return hasNext, nil, err
+		}
+	}
+
+	baseURL := helper.Endpoint.URL
+	queryMetricURL := strings.Replace(AggregatedMetricPath, "{appId}", helper.Client.AppId, -1)
+	queryMetricURL = strings.Replace(queryMetricURL, "{metric_type}", metricName, -1)
+	requestURL := fmt.Sprintf("%s%s", baseURL, queryMetricURL)
+
+	req, err := http.NewRequest("GET", requestURL, nil)
+	req.Header.Add("Authorization", helper.Client.AuthToken)
+	q := req.URL.Query()
+	if startTime > 0 {
+		q.Add("start-time", strconv.FormatInt(startTime, 10))
+	}
+	if endTime > 0 {
+		q.Add("end-time", strconv.FormatInt(endTime, 10))
+	}
+	if desc {
+		q.Add("order", "desc")
+	} else {
+		q.Add("order", "asc")
+	}
+	q.Add("page", strconv.FormatUint(page, 10))
+	req.URL.RawQuery = q.Encode()
+
+	resp, err := helper.DoRequest(req)
+	if err != nil {
+		return hasNext, nil, err
+	}
+	defer resp.Body.Close()
+
+	raw, err := ioutil.ReadAll(resp.Body)
+	if resp.StatusCode != http.StatusOK {
+		var errorMsg string
+		switch resp.StatusCode {
+		case 401:
+			errorMsg = fmt.Sprintf(ui.Unauthorized, baseURL, helper.Client.CCAPIEndpoint)
```

Comment:
do we have token refresh if the access token is expired?

Reply:
Token refresh happens when initializing  CF client and the application context.   Prior to the any policy/metric/history API call to autoscaler,  the cli plugin will invoke cf.GetApp() first in which the token is refreshed.

Reply:
got u